Assets/_Project/Code/Scripts/QuestionStage/AnswerController.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System.Collections;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.SceneManagement;
+using Unity.Netcode;
+using System.Collections.Generic;
+using Unity.Collections;
+using Unity.VisualScripting;
+public class AnswerController : NetworkBehaviour
+{
+    public TMP_InputField answerInput;
+    public TMP_Text questionText;
+    public TMP_Text feedbackText;
+    public TMP_Text timerText;
+    public Button submitButton;
+    public TMP_Text roundNumber;
+    public GameObject hintButtonsContainer;
+    public Button useHintsButton;
+
+    private Button[] answerButtons;
+    public static Team winner;
+    public static int currentQuestionIndex = 0;
+    private float _timeRemaining;
+    private bool _isAnswerChecked;
+
+    public static Category category;
+    public Question currentQuestion;
+
+    void Start()
+    {
+        answerButtons = hintButtonsContainer.GetComponentsInChildren<Button>();
+
+        SetItemsInteractivity(false);
+        currentQuestionIndex++;
+        _isAnswerChecked = false;
+        SetHintMode(false);
+
+        foreach (Button button in answerButtons)
+        {
+            button.onClick.AddListener(() => OnSelectButton(button));
+        }
+
+        if (IsHost)
+        {
+            SetCategoryServerRpc();
+            StartRoundServerRpc();
+        }
+
+        if (Utils.CurrentTeam == winner)
+        {
+            feedbackText.text = "Jesteś graczem ktory wygrał licytacje";
+        }
+
+        else
+        {
+            feedbackText.text = "Jesteś graczem ktory przegrał licytacje. Tryb obserwatora";
+        }
+    }
+
+    private IEnumerator StartCountdown()
+    {
+        while (_timeRemaining > 0 && _isAnswerChecked == false)
+        {
+            _timeRemaining -= Time.deltaTime;
+            ShowCurrentTimeRpc(_timeRemaining);
+            yield return null;
+        }
+        if (_timeRemaining <= 0)
+        {
+            SetItemsInteractivity(false);
+            feedbackText.text = "Czas minął! Odpowiedzi: " + string.Join(", ", currentQuestion.CorrectAnswers);
+            StartCoroutine(ChangeScene("Lobby", 4));
+        }
+    }
+
+    void StartServerTime()
```

Comment:
Czy ta funkcja jest w ogóle potrzebna? Wywołuje tylko inną funkcje

Reply:
Z tego co pamiętam to nie da się wywołać enumeratora w funkcji rpc. Dlatego funkcja rpc wywołuje voida który wywołuje enumerator

Reply:
Jak wkleiłem StartCoroutine do StartRoundServerRpc to chyba wszystko działało, ale może nie jestem czegoś świadomy